monkey/monkey_island/setup/island_config_options.py

```diff
@@ -0,0 +1,30 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from monkey_island.cc.server_utils.consts import (
+    DEFAULT_DATA_DIR,
+    DEFAULT_LOG_LEVEL,
+    DEFAULT_START_MONGO_DB,
+)
+
+
+@dataclass
+class IslandConfigOptions:
+    log_level = DEFAULT_LOG_LEVEL
+    data_dir = DEFAULT_DATA_DIR
+    start_mongodb = DEFAULT_START_MONGO_DB
+
+    @staticmethod
+    def build_from_config_file_contents(config_contents: dict) -> IslandConfigOptions:
+        config = IslandConfigOptions()
+        if "data_dir" in config_contents:
```

Comment:
Looking at the documentation, I don't think this is properly using the `@dataclass` decorator. https://docs.python.org/3/library/dataclasses.html

Reply:
Added type hints, though they are not mandatory.